src/map.rs

```diff
@@ -95,6 +95,7 @@ where
     }
 }
 
+#[cfg(feature = "std")]
```

Comment:
I wonder if we could play the same trick here of using `where S: Default`?

Reply:
We can! The only reason we were locked to `RandomState` was that we needed a method for initializing the `BuildHasher`.

Reply:
Done.